From david-b@pacbell.net Tue Dec  5 10:09:43 2006
From: David Brownell <david-b@pacbell.net>
To: linux-pm@lists.osdl.org
Subject: Driver core: deprecate PM_LEGACY, default it to N
Date: Tue, 5 Dec 2006 03:07:38 -0800
Cc: Greg KH <greg@kroah.com>
Message-Id: <200612050307.39157.david-b@pacbell.net>

Deprecate the old "legacy" PM API, and more importantly default it to "n".
Virtually nothing in-tree uses it any more.

Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 kernel/power/Kconfig |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

--- gregkh-2.6.orig/kernel/power/Kconfig
+++ gregkh-2.6/kernel/power/Kconfig
@@ -20,13 +20,14 @@ config PM
 	  sending the processor to sleep and saving power.
 
 config PM_LEGACY
-	bool "Legacy Power Management API"
+	bool "Legacy Power Management API (DEPRECATED)"
 	depends on PM
-	default y
+	default n
 	---help---
-	   Support for pm_register() and friends.
+	   Support for pm_register() and friends.  This old API is obsoleted
+	   by the driver model.
 
-	   If unsure, say Y.
+	   If unsure, say N.
 
 config PM_DEBUG
 	bool "Power Management Debug Support"
